pkg/config/autodiscovery/autodiscovery_test.go

```diff
@@ -44,5 +44,19 @@ snmp_listener:
 `))
 	assert.NoError(t, err)
 	_, configListeners = DiscoverComponentsFromConfig()
-	assert.Empty(t, len(configListeners))
+	assert.Len(t, configListeners, 1)
+	assert.Equal(t, "snmp", configListeners[0].Name)
```

Comment:
This follows the same pattern as what is done above in the same file.
Yet, it is problematic as it won’t skip the evaluation of `configListeners[0]` when `configListeners` is empty.
Indeed, `assert.XXX(…)` will eventually make the whole test fail, but it doesn’t abort its execution prematurely.
On the other hand, `require.XXX(…)` makes the test abort prematurely and skips the execution of the rest of it.

So, in order to avoid panicking when `configListeners` is empty, we shoud either do:
```suggestion
	require.Len(t, configListeners, 1)
	assert.Equal(t, "snmp", configListeners[0].Name)
```

or, if we don’t want to skip the execution of the rest of the test,
```suggestion
	if assert.Len(t, configListeners, 1) {
		assert.Equal(t, "snmp", configListeners[0].Name)
	}
```